### DISCUSSION TITLE
Reusable functions for custom endpoints

### DISCUSSION THREAD

**Discussion by thebaron2** [directus/directus#7309] [Archive]
Hi there,
Any suggestions on a folder structure for functions that different custom endpoints can use?

**Comment by rijkvanzanten** [comment]
In a separate file! Custom endpoints run in node, so you can simply use Node's `require` to import modular code 👍🏻

**Reply by thebaron2** [reply]
thanks for the quick reply, but do you have any tips or suggestions on how such a folder structure would look like? The custom endpoints are in the endpoints folder of course.

I realize i should have been clearer in my original question, will edit that.

**Reply by rijkvanzanten** [reply]
So under extensions, you'll have your main extensions folder:

```
endpoints/
  my-custom-endpoint/
    index.js // << Holds the router, registers the routes
    my-handler-function.js
    my-other-handler-function.js
```

Then in the main index.js file (the one that's read by directus), you can import your other functions using `require`:

```js
// index.js

const handlerFunction = require('./my-handler-function');
const otherHandlerFunction = require('./my-other-handler-function');

module.exports = function registerRoutes(router) {
  router.get('/', handlerFunction);
  router.get('/other', otherHandlerFunction);
}
```

```js
// my-handler-function.js

module.exports = function(req, res) {
  res.send('Hello World!');
}
```

**Reply by shashwattpandeyy** [reply]
How can we pass services as a parameter in my-handler-function? So that I have access to the database

**Reply by nickrum** [reply]
You could make the function exported from `my-handler-function.js` a function that returns another function and pass the service to it:

```js
function(services) {
  return (req, res) => {

    // use services here

    res.send('Hello World!');
  };
}
```
